feat(components/indicators): change alertType to SkyIndicatorIconType

Blackbaud-SandhyaRajasabeson opened the pull request:
[AB#2195283](https://blackbaud.visualstudio.com/Products/_boards/board/t/SKY%20UX%20Components/Stories/?workitem=2195283)

- changes `alertType` from string to `SkyIndicatorIconType` 
- adds alert playground code to mono repo
- adds examples for default and invalid `alertType` to playground code
BREAKING CHANGE: This change removes support for `alertType` on the alert component being an
unaccepted string. To address this change, change the `alertType` to an accepted `SkyIndicatorTypeIcon` or
remove it to use the default `alertType` of `'warning'`.

BEGIN_COMMIT_OVERRIDE
feat(components/indicators): change `alertType` to `SkyIndicatorIconType` (#683)

BREAKING CHANGE: This change removes support for `alertType` on the alert component being an
unaccepted string. To address this change, change the `alertType` to an accepted `SkyIndicatorTypeIcon` or
remove it to use the default `alertType` of `'warning'`.
END_COMMIT_OVERRIDE

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/blackbaud/skyux?pullRequest=683) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/blackbaud/skyux?pullRequest=683) before we can accept your contribution.<br/>

blackbaud-sky-savage commented:
**Tests failed.** Automated cross-browser testing via BrowserStack and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: 508501111f0ef172b7ffed5cf1dde5d88ebbf23a
Build details: https://travis-ci.org/blackbaud-sky-savage/skyux/builds/143909129

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

Blackbaud-JasonRikard commented:
[plnkr](http://plnkr.co/edit/XU8FTCY1IKXwOYkPsW6P?p=preview) with grid with addJsonData in the loadMore()

Blackbaud-PatrickOFriel commented:
Looks like there are test failures @Blackbaud-JasonRikard. Do they fail locally as well?

Blackbaud-JasonRikard commented:
Yeah, they fail locally on grunt test. Working on a resolution.

Blackbaud-JasonRikard commented:
I reworked the tests but I ended up with a version of the previous solution where the table needed reinitialization. I'm closing this in favor of building an object array of rows rather than trying to get addJsonData() to be additive.

Commit:
feat(components/indicators/alert): change alertType from string to SkyIndicatorIconType

breaking change in preparation for strict mode. Change alertType from string to SkyIndicatorIconType

BREAKING CHANGE: This change removes support for `alertType` on the alert component  being an
unaccepted string. To address this change the `alertType` to an accepted `SkyIndicatorTypeIcon` or
remove it to use the default alert.

Commit:
remove todo

blackbaud-sky-build-user commented:
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/683/storybook/)

Component Storybooks:

- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/683/storybooks/ag-grid-storybook/)
- [indicators-storybook](https://blackbaud.github.io/skyux-pr-preview/683/storybooks/indicators-storybook/)
- [layout-storybook](https://blackbaud.github.io/skyux-pr-preview/683/storybooks/layout-storybook/)
- [modals-storybook](https://blackbaud.github.io/skyux-pr-preview/683/storybooks/modals-storybook/)
- [pages-storybook](https://blackbaud.github.io/skyux-pr-preview/683/storybooks/pages-storybook/)
- [popovers-storybook](https://blackbaud.github.io/skyux-pr-preview/683/storybooks/popovers-storybook/)
- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/683/storybooks/tabs-storybook/)
- [tiles-storybook](https://blackbaud.github.io/skyux-pr-preview/683/storybooks/tiles-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/683/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/683/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/683/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/683/playground/)

Commit:
playground icon
